src/script/components/list/ParticipantItem.tsx

```diff
@@ -89,36 +87,43 @@ const ParticipantItem: React.FC<ParticipantItemProps> = ({
   const hasUsernameInfo = isUser && !hideInfo && !hasCustomInfo && !isTemporaryGuest;
   const isOthersMode = mode === UserlistMode.OTHERS;
 
-  const isFederated = participant instanceof User && participant.isFederated;
-  const isGuest = participant instanceof User && !isFederated && useKoSubscribable(participant.isGuest);
-  const isVerified = useKoSubscribable((participant as User).is_verified ?? ko.observable(false));
-  const availability = useKoSubscribable((participant as User).availability ?? ko.observable<Availability.Type>());
-
-  const participantName = useKoSubscribable(
-    isUser ? (participant as User).name : ko.observable((participant as ServiceEntity).name),
-  );
-  const callParticipantSharesCamera = useKoSubscribable(callParticipant?.sharesCamera ?? ko.observable(false));
-  const callParticipantSharesScreen = useKoSubscribable(callParticipant?.sharesScreen ?? ko.observable(false));
-  const callParticipantIsActivelySpeaking = useKoSubscribable(
-    callParticipant?.isActivelySpeaking ?? ko.observable(false),
+  const {
+    is_verified: isVerified,
+    isGuest,
+    availability,
+    expirationText,
+    name: participantName,
+  } = useKoSubscribableChildren(
+    // We need to make TS believe that this is a User, otherwise it will complain about 
+    // the fields in the array that don't exist on ServiceEntity
+    participant as User,
+    participant instanceof User ? ['isGuest', 'is_verified', 'availability', 'expirationText', 'name'] : ['name'],
   );
 
-  const callParticipantIsMuted = useKoSubscribable(callParticipant ? callParticipant.isMuted : ko.observable());
+  const isFederated = participant instanceof User && participant.isFederated;
```

Comment:
Instead of this, you could replace `isGuest` by `isDirectGuest` that handles the federation case under the hood ;)